src/tests/system/tests/test_ipa.py

```diff
@@ -640,3 +641,513 @@ def test_ipa__switch_user_with_smartcard_authentication(client: Client, ipa: IPA
+    for user in users:
+        ipa.user(user).add()
+    remote_svc_group = ipa.hbacsvcgroup("remote_access").add(description="Remote access services")
+    remote_svc_group.add_member(hbacsvc=["sshd", "login"])
+    ipa.hbac("allow_all").disable()
+    ipa.hbac("remote_services_access").create(
+        description="Remote access via service groups",
+        users="user1",
+        hosts="client.test",
+        servicegroups="remote_access",
+    )
+
+    client.sssd.restart()
+    assert client.auth.ssh.password("user1", "Secret123"), "user1 SSH should succeed!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "user2 SSH should be denied!"
+
+    remote_svc_group.remove_member(hbacsvc=["sshd"])
+    client.sssd.restart()
+    assert not client.auth.ssh.password(
+        "user1", "Secret123"
+    ), "user1 SSH should be denied after sshd removal from service group!"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__hbac_multiple_rules_match_priority(client: Client, ipa: IPA):
+    """
+    :title: Validate HBAC rule priority when multiple rules match
+    :setup:
+      1. Add users
+      2. Disable default HBAC rule
+      3. Create two HBAC rules for a user
+    :steps:
+      1. Restart SSSD and validate SSH login for users
+      2. Disable first rule; restart SSSD, verify user's access
+    :expectedresults:
+      1. User allowed access via either rule
+      2. Successfully disabled and SSSD restarts, user1SSH login succeeds via second rule after disabling first
+    :customerscenario: False
+    """
+    users = ["user1", "user2"]
+    for user in users:
+        ipa.user(user).add()
+    ipa.hbac("allow_all").disable()
+    primary_user1_ssh_rule = ipa.hbac("primary_user1_ssh").create(
+        description="Primary SSH access rule for user1", users="user1", hosts="client.test", services="sshd"
+    )
+    ipa.hbac("secondary_user1_ssh").create(
+        description="Secondary SSH access rule for user1", users="user1", hosts="client.test", services="sshd"
+    )
+
+    client.sssd.restart()
+    assert client.auth.ssh.password("user1", "Secret123"), "user1 SSH should succeed!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "user2 SSH should be denied!"
+
+    primary_user1_ssh_rule.disable()
+    client.sssd.restart()
+    assert client.auth.ssh.password("user1", "Secret123"), "user1 SSH should still work via second rule!"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__hbac_category_all_users(client: Client, ipa: IPA):
+    """
+    :title: Validate HBAC rule with userCategory='all'
+    :setup:
+      1. Add users.
+      2. Disable default HBAC rule
+      3. Create HBAC rule with userCategory='all'
+    :steps:
+      1. Restart SSSD and verify SSH login for all users
+      2. Modify rule to remove userCategory='all' and allow only a user, restart SSSD and verify access for a user only
```

Comment:
I'm pretty certain this is over 120 characters? Nope, exactly at 120, but lets break this up into two.